fs/ftpfs.py

```diff
@@ -235,6 +235,13 @@ def read(self, size=-1):
                 remaining -= len(chunk)
         return b"".join(chunks)
 
+    def readinto(self, buffer):
+        # type: (bytearray) -> int
+        data = self.read(len(buffer))
+        bytes_read = len(data)
+        buffer[: len(data)] = data
```

Comment:
Any reason this isn't `buffer[:bytes_read] = data` ?

Reply:
Ahh, good point.  I just copied the implementation from `fs.iotools.RawWrapper.readinto`.  The implementation for `fs.iotools.RawWrapper.readinto1` is also the same.  I could update all three locations if requested. Cheers!

Reply:
@nwh : could you please update this (as well as the code in `RawWrapper.readinto`)? I'll merge the PR right after. Thanks!

Reply:
@althonos done -- however, there is conflict in the change log.  How would you like that handled?